### PR TITLE
feat: support subpage

### PR DESCRIPTION
**Pull request by QiShaoXuan**

### CODE REVIEW

**Comment by himself65** [comment]
https://github.com/toeverything/AFFiNE/actions/runs/4496996815

File: apps/web/src/components/pure/workspace-slider-bar/pivot/Pivot.tsx [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+
+  const treeData = useMemo(
+    () => flattenToTree(allMetas, { openPage }),
+    [allMetas, openPage]
+  );
+
+  const handleAdd = useCallback(
+    (node: TreeNode) => {
+      const id = nanoid();
+      // FIXME: remove ts-ignore after blocksuite update
```
Comment (himself65) [inline apps/web/src/components/pure/workspace-slider-bar/pivot/Pivot.tsx]: remove?

File: apps/web/src/components/pure/workspace-slider-bar/pivot/Pivot.tsx [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+          logger.info('drop into root and resort');
+
+          if (dragParentMeta) {
+            const newSubpageIds = [...(dragParentMeta.subpageIds ?? [])];
+
+            const deleteIndex = dragParentMeta.subpageIds?.findIndex(
+              id => id === dragNode.id
+            );
+            newSubpageIds.splice(deleteIndex, 1);
+            // FIXME: remove ts-ignore after blocksuite update
```
Comment (himself65) [inline apps/web/src/components/pure/workspace-slider-bar/pivot/Pivot.tsx]: remove

File: apps/web/src/components/pure/workspace-slider-bar/pivot/Pivot.tsx [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+        // FIXME: remove ts-ignore after blocksuite update
+        // @ts-ignore
+        setPageMeta(dragParentMeta.id, {
+          subpageIds: newSubpageIds,
+        });
+      }
+      const dropMeta = allMetas.find(meta => meta.id === dropNode.id)!;
+      const newSubpageIds = [dragNode.id, ...(dropMeta.subpageIds ?? [])];
+      // FIXME: remove ts-ignore after blocksuite update
+      // @ts-ignore
```
Comment (himself65) [inline apps/web/src/components/pure/workspace-slider-bar/pivot/Pivot.tsx]: remove

**Review by himself65 — COMMENTED** [review]
LGTM but need @darkskygit to confirm that the Cloudflare build is ok for new code

**Comment by darkskygit** [comment]
> LGTM but need @darkskygit to confirm that the Cloudflare build is ok for new code

fixed

**Review by darkskygit — APPROVED** [review]

**Pull request merged**